src/main/kotlin/ar/edu/unahur/obj2/socios/Cliente.kt

```diff
@@ -1,4 +1,45 @@
 package ar.edu.unahur.obj2.socios
 
-class Cliente {
+
+object Cliente { //Patron Singleton aplicado
+    lateinit var estadoDeAnimo: EstadoDeAnimo //Los atributos del creador fueron cambiados por 'lateinit var' ya que se van a declarar cuando sea necesario
+    var bolsillo: Double = 0.0
+    lateinit var pedido : Pedido
+    lateinit var barrio: Barrio
+
+    //Getters
+    fun consultarPropina() = this.estadoDeAnimo.propina(this)
+    fun consultarPropinaBarrio() = this.barrio.cuenta(this.consultarPropina())
+
+    //Setters
+    fun hacerPedido(comida: Pedido){ pedido = comida }
+    fun cambiarBarrio(nuevoBarrio: Barrio) { barrio = nuevoBarrio }
+    fun cambiarEstadoDeAnimo(nuevoEstado: EstadoDeAnimo) { estadoDeAnimo = nuevoEstado }
+    fun cambiarBolsillo(nuevoBolsillo: Double) { bolsillo = nuevoBolsillo }
+}
+
+interface EstadoDeAnimo {
+    fun propina(cliente: Cliente) : Double
+}
```

Comment:
El estado de animo es un atributo del Cliente, por lo que tiene sentido que reciba a el Cliente, pero no tendría que poder acceder a los atributos del pedido, una forma más sencilla de calcular el precio es que lo reciba directamente por parametro además del cliente.

```suggestion
interface EstadoDeAnimo {
    fun propina(Double: precioPedido, Cliente: cliente) : Double
}

object Enojado : EstadoDeAnimo {
    override fun propina(Double: precioPedido, , Cliente: cliente) = 0.0 
}

object Feliz : EstadoDeAnimo {
    override fun propina(cliente: Cliente, Cliente: cliente) = precioPedido * 0.25 
}
```